hal_st/stm32fxxx/TimerStm.hpp

```diff
@@ -46,16 +46,17 @@ namespace hal
         TimerBaseStm(uint8_t aTimerIndex, Timing timing, const Config& config);
         ~TimerBaseStm();
 
-        void Start();
-        void Stop();
+        virtual void Start();
```

Comment:
In what way does it help you that Start() and Stop() are now virtual? Wouldn't you rather derive TimerBaseStm from an interface?

Reply:
In this particular case I want to reuse how a Timer peripheral is initialized but want to change the start/stop behavour (IRQ mode). I can also create an interface for a peripheral Timer if that's preferred (in `embedded-infra-lib/hal/interfaces` ?).

Reply:
Perhaps, although breaking, it then makes sense to make TimerBaseStm abstract, and derive concrete versions: `FreeRunningTimerStm` `ImmediateRolloverTimerStm` `DispatchedRolloverTimerStm` for example.

Reply:
I implemented this class in order to use together with ADC and DMA. At that moment I didn't see a need to create a specific interface for that. AdcWithDma class is possibly the only implementation using it.

Reply:
I'll create a separate PR for the `TimerBaseStm`-related changes.